src/KeyringController.ts

```diff
@@ -258,6 +258,29 @@ export class KeyringController extends BaseController<BaseConfig, KeyringState>
 		return this.keyring.signPersonalMessage(messageParams);
 	}
 
+	/**
+	 * Signs typed message by calling down into a specific keyring
+	 *
+	 * @param messageParams - MessageParams object to sign
+	 * @param version - Compatibility version EIP712
+	 * @returns - Promise resolving to a signed message string or an error if any
+	 */
+	async signTypedMessage(messageParams: MessageParams, version: string) {
+		try {
+			const address = sigUtil.normalize(messageParams.from);
+			const privateKey = await this.exportAccount(address);
+			const privateKeyBuffer = ethUtil.toBuffer(ethUtil.addHexPrefix(privateKey));
+			switch (version) {
+				case 'V1':
+					return sigUtil.signTypedDataLegacy(privateKeyBuffer, { data: messageParams.data });
+				case 'V3':
```

Comment:
should we make version optional and default to v3?

Reply:
good idea, what do you think @bitpshr ? This was the way it is [here](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/typed-message-manager.js)